src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Device.Gpio;
+using System.Device.Spi;
+
+namespace Iot.Device.Spi
+{
+    public class SoftwareSpi : SpiDevice
+    {
+        private GpioController _controller;
+        private readonly int _clk;
+        private readonly int _miso;
+        private readonly int _mosi;
+        private readonly int _cs;
+        private readonly SpiConnectionSettings _settings;
+
+        public SoftwareSpi(int clk, int miso, int mosi, int cs, SpiConnectionSettings settings = null, GpioController controller = null)
```

Comment:
Can you please add XML docs to all of these? We are now packing the docs in the package so it will be very useful for people developing on VSCode or Visual Studio to get info on what each of this parameters should be.